tests/integration_testing/IntegrationTesting.hs

```diff
@@ -0,0 +1,163 @@
+    ( bazelCmd
+    , setupWorkspace
+    , setupTestBazel
+    , assertSuccess
+    , assertFailure
+    , outputSatisfy
+    , failedOutputSatisfy
+    , formatOutput
+    ) where
+
+import qualified System.Process as Process
+import System.Environment (getEnv, getArgs, unsetEnv)
+import System.Info (os)
+import System.FilePath (pathSeparators, (</>))
+import System.Directory (copyFile, doesDirectoryExist, doesFileExist, removePathForcibly, createDirectory, listDirectory, doesPathExist)
+import Data.Text (pack, unpack, replace, breakOn)
+import Data.Text.IO (readFile, writeFile)
+import System.Exit (ExitCode(..))
+import Control.Monad (when, unless, forM_)
+import Test.Hspec (shouldSatisfy, expectationFailure)
+
+bazelCmd :: String -> String -> IO ([String] -> Process.CreateProcess)
+bazelCmd workspaceDir outputUserRoot = do
+    bazelPath <- getEnv "BIT_BAZEL_BINARY"
+    config <- (fmap bazelConfig isNixpkgs)
+    return (\args -> case args of
+        subcommand:xs | elem subcommand ["build", "test", "run", "query"] -> (Process.proc bazelPath (["--output_user_root", outputUserRoot, subcommand, "--config", config] ++ xs)) { Process.cwd = Just workspaceDir }
+        xs -> (Process.proc bazelPath (["--output_user_root", outputUserRoot] ++ xs)) { Process.cwd = Just workspaceDir })
+
+isNixpkgs = (elem "nixpkgs") <$> getArgs
+
+bazelConfig isnix
+    | isnix = case os of
+        "darwin" -> "macos-nixpkgs"
+        _ -> "linux-nixpkgs"
+    | otherwise = case os of
+        "darwin" -> "macos-bindist"
+        "mingw32" -> "windows-bindist"
+        _ -> "linux-bindist"
+
+outputBaseDir = do
+    tmpDir <- getEnv "TEST_TMPDIR"
+    return (unpack . fst $ breakOn (pack (pathSeparators ++ "execroot" ++ pathSeparators)) (pack tmpDir))
+
+replaceInFile :: FilePath -> String -> String -> IO ()
+replaceInFile path from to = do
+    content <- Data.Text.IO.readFile path
+    Data.Text.IO.writeFile path (replace (pack from) (pack to) content)
+
+removeDirIfExist path = do
+    dirExist <- doesDirectoryExist path
+    when dirExist (removePathForcibly path)
+
+createDirIfNotExist path = do
+    dirExist <- doesDirectoryExist path
+    unless dirExist (createDirectory path)
+
+generateBazelRc dir = do
+    alreadyExist <- doesFileExist (dir </> ".bazelrc")
+    unless alreadyExist $ Data.Text.IO.writeFile (dir </> ".bazelrc") (pack " \n\
+\ build:linux-nixpkgs --host_platform=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host \n\
+\ build:linux-nixpkgs --incompatible_enable_cc_toolchain_resolution \n\
+\ build:macos-nixpkgs --host_platform=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host \n\
+\ build:macos-nixpkgs --incompatible_enable_cc_toolchain_resolution \n\
+\ build:linux-bindist --incompatible_enable_cc_toolchain_resolution \n\
+\ build:macos-bindist --incompatible_enable_cc_toolchain_resolution \n\
+\ build:windows-bindist --crosstool_top=@rules_haskell_ghc_windows_amd64//:cc_toolchain \n\
+\ ")
+
+setupWorkspace :: IO (String, String)
```

Comment:
I think a docstring that describes the workspace setup and how rules_haskell is important would be useful

Reply:
I was thinking about put a README.md in tests/integration_tesinting/ forlder

Reply:
Done